format/spec.md

```diff
@@ -631,6 +632,30 @@ When expiring snapshots, retention policies in table and snapshot references are
     2. The snapshot is not one of the first `min-snapshots-to-keep` in the branch (including the branch's referenced snapshot)
 5. Expire any snapshot not in the set of snapshots to retain.
 
+#### Statistics file
+
+Statistics files are valid [Puffin files](../puffin-spec). Statistics are informational. A reader can choose to
+ignore statistics information. Statistics support is not required to read the table correctly.
+
+Statistics file's metadata within `statistics` table snapshot field is a struct with the following fields:
+
+| v1         | v2         | Field name                      | Type                              | Description                                                                                                                               |
+|------------|------------|---------------------------------|-----------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------|
+| _required_ | _required_ | **`statistics-path`**           | `string`                          | Path of the statistics file. See [Puffin file format](../puffin-spec).                                                                    |
+| _required_ | _required_ | **`file-size-in-bytes`**        | `long`                            | Size of the statistics file.                                                                                                              |
```

Comment:
Do we need an encryption key_metadata here as in the DataFile Spec?

Reply:
I'm not sure our security review team will agree. Something like first and last name combos which have a very finite set of values, the ability to detect whether the dataset contains a value +- a false positive rate may be considered a vulnerability. I'm gonna see if they'll accept it.

Reply:
+1 to adding encryption key_metadata here, similar to the ManifestFile structures (we can also use the same approach to their encryption). 
Regarding sensitivity of this metadata - as an example, Parquet encrypts the bloom filters. Here goes the doc section with details (and reasons) 
https://github.com/apache/parquet-format/blob/master/BloomFilter.md#encryption

Reply:
Sounds reasonable.

Reply:
Sure, i can add `key_metadata | Base64-encoded implementation-specific key metadata for encryption` here.

do we want whole-file encryption, or do we need to encrypt individual blobs?
Also, if a file is encrypted, is it more useful to store file-size-in-bytes before encryption, or we also/instead need to have a number after the encryption?

Reply:
> add key_metadata | Base64-encoded implementation-specific key metadata for encryption here.

sgtm

> do we want whole-file encryption, or do we need to encrypt individual blobs?

Both key_metadata and the encryption itself are implementation-specific, so probably no need to add these details here. (wrt implementations, we can start with a simple file encrypting stream, similar to manifest files; and expand to other techniques later)

> if a file is encrypted, is it more useful to store file-size-in-bytes before encryption, or we also/instead need to have a number after the encryption

this depends on the intended use of this field. For manifest files, we don't use the file-size-in-bytes field for decryption - we get the full file length directly from InputFile.getLength() which is sourced from the file system. In other words, decryption is ok with any definition of this field (because it doesn't use it).

Reply:
@rdblue are you OK with me adding the following

`add key_metadata | Base64-encoded implementation-specific key metadata for encryption here.`

within this PR?

Reply:
Yes, I think it is a reasonable addition.